Allow presenting a subclass of the current model context

kobsy opened the pull request:
One more fix! 😝 Model checking is great, but it currently doesn't handle subclasses. In looking into this, I learned that the `<=` operator works when comparing classes & modules in Ruby ... so it was actually a pretty straightforward change to enable it!

I wanted to write a test without completely reworking the test schema you'd come up with for testing relationships, so I added `Periodical` and `Journal`, though if you'd rather integrate STI into your existing schema, I'd completely understand.

boblail commented on test/support/schema.rb:
```diff
@@ -28,9 +28,9 @@
 
-  create_table "people", :force => true do |t|
-    t.string :first_name
-    t.string :last_name
```
Oh! I duplicated the people table ... 🤦‍♂

boblail replied:
I pushed a commit to fix that.

boblail commented:
👍 Nice catch! 

TIL about `<=` 😄 

What if we add this column to `books` instead of a creating a new table?

```ruby
  t.string :type, null: false, default: "Book"
```

(and you could add something other than Novel as a subclass of Book — like Biography or whatever you feel like)

(I was working hard to keep the schema tight — partially to keep this 👇 from growing and growing and growing!)

```
-- create_table("books", {:force=>true})
   -> 0.1162s
-- create_table("books_characters", {:id=>false, :force=>true})
   -> 0.0074s
-- create_table("characters", {:force=>true})
   -> 0.0051s
-- create_table("isbns", {:force=>true})
   -> 0.0055s
-- create_table("people", {:force=>true})
   -> 0.0061s
```

kobsy commented:
I updated the tests to add the `type` column to `books` as you suggested and just went with "Novel" as my subclass. I also rebased, so we should be good there. Ready for another look!

boblail commented:
Matt, do you want to make a PR of this commit against the branch `v0.6.x` as well? (It'll have to be tweaked a little)

kobsy commented:
Neat! Just created #19, which hopefully is set up right to backport the fix. 👍
